### PR TITLE
Catch2 Test Fixes

### PR DESCRIPTION
**Pull request by pvelesko**
@gargrahul created a new PR as requested.

### CODE REVIEW

**Comment by pvelesko** [comment]
@gargrahul

**Comment by gargrahul** [comment]
@pvelesko  Could you please rebase your branch with latest develop? Also please update Unit_hipEvent.cc after Line # 141 with
HIP_ASSERT(hipGetLastError() == hipErrorInvalidHandle);
This should fix the failure seen on Nvidia backend.

**Comment by pvelesko** [comment]
@gargrahul sure give me a minute.

**Comment by pvelesko** [comment]
@gargrahul I need to make this change because this PR is based on my fork, correct? And if you were to clone it and make it an AMD branch then I wouldn't be able to push changes?

**Comment by rorake** [comment]
In below testcase files 'HIP_CHECK(hipGetLastError());' is not added after hipLaunchKernelGGL calls. 
Not sure if this needs to be incorporated in this PR or different PR.

tests/catch/multiproc/hipIpcEventHandle.cc

tests/catch/unit/cooperativeGrps/coalesced_groups_shfl_down.cc:  
tests/catch/unit/cooperativeGrps/coalesced_groups_shfl_up.cc:
tests/catch/unit/cooperativeGrps/simple_coalesced_groups.cc:
 
tests/catch/unit/deviceLib/AtomicAdd_Coherent_withnoUnsafeflag.cc
tests/catch/unit/deviceLib/AtomicAdd_Coherent_withoutflag.cc
tests/catch/unit/deviceLib/AtomicAdd_Coherent_withunsafeflag.cc
tests/catch/unit/deviceLib/AtomicAdd_NonCoherent_withnoUnsafeflag.cc
tests/catch/unit/deviceLib/AtomicAdd_NonCoherent_withoutflag.cc 
tests/catch/unit/deviceLib/AtomicAdd_NonCoherent_withunsafeflag.cc 
tests/catch/unit/deviceLib/BuiltIns_fadd.cc
tests/catch/unit/deviceLib/BuiltIns_fmax.cc
tests/catch/unit/deviceLib/BuiltIns_fmin.cc
tests/catch/unit/deviceLib/hipTestDeviceSymbol.cc
tests/catch/unit/deviceLib/unsafeAtomicAdd_Coherent_withnounsafeflag.cc
tests/catch/unit/deviceLib/unsafeAtomicAdd_Coherent_withoutflag.cc 
tests/catch/unit/deviceLib/unsafeAtomicAdd_Coherent_withunsafeflag.cc 
tests/catch/unit/deviceLib/unsafeAtomicAdd_NonCoherent_withnounsafeflag.cc 
tests/catch/unit/deviceLib/unsafeAtomicAdd_NonCoherent_withoutflag.cc
tests/catch/unit/deviceLib/unsafeAtomicAdd_NonCoherent_withunsafeflag.cc 

tests/catch/unit/memory/hipMallocPitch.cc
tests/catch/unit/memory/hipPointerGetAttribute.cc

tests/catch/unit/multiThread/hipMultiThreadStreams1.cc
tests/catch/unit/multiThread/hipMultiThreadStreams2.cc

tests/catch/unit/texture/hipBindTex2DPitch.cc
tests/catch/unit/texture/hipBindTexRef1DFetch.cc
tests/catch/unit/texture/hipNormalizedFloatValueTex.cc
tests/catch/unit/texture/hipSimpleTexture2DLayered.cc
tests/catch/unit/texture/hipSimpleTexture3D.cc
tests/catch/unit/texture/hipTex1DFetchCheckModes.cc
tests/catch/unit/texture/hipTexObjPitch.cc
tests/catch/unit/texture/hipTextureMipmapObj2D.cc
tests/catch/unit/texture/hipTextureObj1DCheckModes.cc 
tests/catch/unit/texture/hipTextureObj1DCheckSRGBModes.cc
tests/catch/unit/texture/hipTextureObj1DFetch.cc
tests/catch/unit/texture/hipTextureObj2D.cc
tests/catch/unit/texture/hipTextureObj2DCheckModes.cc 
tests/catch/unit/texture/hipTextureObj2DCheckSRGBModes.cc
tests/catch/unit/texture/hipTextureObj3DCheckModes.cc
tests/catch/unit/texture/hipTextureObjFetchVector.cc
tests/catch/unit/texture/hipTextureRef2D.cc

**Comment by pvelesko** [comment]
I'll add them here now

**Comment by pvelesko** [comment]
@rorake  please verify.

File: tests/catch/unit/cooperativeGrps/coalesced_groups_shfl_up.cc [hunk 1]
```diff
@@ -207,6 +208,7 @@ static void test_shfl_up() {
     hipLaunchKernelGGL(kernel_shfl_up, blockSize, threadsPerBlock,
                        threadsPerBlock * sizeof(int), 0, dPtr, dResults, lane_delta, i);
     hipMemcpy(hPtr, dResults, group_size_in_bytes, hipMemcpyDeviceToHost);
+    HIP_CHECK(hipGetLastError());
```
Comment (rorake) [inline tests/catch/unit/cooperativeGrps/coalesced_groups_shfl_up.cc:211]: Shouldn't this HIP_CHECK(hipGetLastError()) be present before the hipMemcpy? That's how it has been done in line # 141.

File: tests/catch/unit/texture/hipSimpleTexture2DLayered.cc [hunk 2]
```diff
@@ -89,6 +89,7 @@ TEST_CASE("Unit_hipSimpleTexture2DLayered_Check") {
   for (unsigned int layer = 0; layer < num_layers; layer++)
     hipLaunchKernelGGL(simpleKernelLayeredArray, dimGrid, dimBlock, 0, 0,
                        dData, width, height, layer);
+  HIP_CHECK(hipGetLastError());
```
Comment (rorake) [inline tests/catch/unit/texture/hipSimpleTexture2DLayered.cc]: Should this HIP_CHECK(hipGetLastError()) be present inside the for loop?

**Comment by rorake** [comment]
tests/catch/unit/deviceLib/BuiltIns_fmin.cc, line # 113 : Please add HIP_CHECK(hipGetLastError())

**Review by gargrahul — APPROVED** [review]

**Pull request merged**